KubeArmor/BPF/enforcer.bpf.c

```diff
@@ -620,16 +620,128 @@ int BPF_PROG(enforce_net, struct socket *sock, struct sockaddr *address,
 
   if (allow) {
     if (!match) {
-      bpf_printk("denying sock type %d, family %d, protocol %d due to not in "
+      bpf_printk("denying sock connect - type %d, family %d, protocol %d due "
+                 "to not in "
                  "allowlist \n",
                  sock->type, address->sa_family, sock->sk->sk_protocol);
       return -EPERM;
     }
   } else {
     if (match) {
-      bpf_printk("denying sock type %d, family %d, protocol %d due to in "
+      bpf_printk(
+          "denying sock connect - type %d, family %d, protocol %d due to in "
+          "blacklist \n",
+          sock->type, address->sa_family, sock->sk->sk_protocol);
+      return -EPERM;
+    }
+  }
+  return 0;
+}
+
+SEC("lsm/socket_accept")
+int BPF_PROG(enforce_net_accept, struct socket *sock) {
+  struct task_struct *t = (struct task_struct *)bpf_get_current_task();
+
+  bool match = false;
+
+  struct outer_key okey = {.pid_ns = get_task_pid_ns_id(t),
+                           .mnt_ns = get_task_mnt_ns_id(t)};
+
+  if (okey.pid_ns == PROC_PID_INIT_INO) {
+    okey.pid_ns = 0;
+    okey.mnt_ns = 0;
+  }
```

Comment:
maybe we should have an static inline function to get `okey` and use this in all places.
```suggestion
static inline void get_outer_key(struct outer_key *pokey) { //pokey = pointer to okey
  *pokey = {.pid_ns = get_task_pid_ns_id(t),
                   .mnt_ns = get_task_mnt_ns_id(t)};

  if (pokey->pid_ns == PROC_PID_INIT_INO) {
    okey->pid_ns = 0;
    okey->mnt_ns = 0;
  }
 }
```